com.unity.uiwidgets/Runtime/painting/colors.cs

```diff
@@ -12,13 +12,13 @@ public partial class painting_ {
             if (max == 0.0f || delta == 0.0f) {
                 hue = 0.0f;
             }
-            else if (max == red) {
+            else if (Math.Abs(max - red) < foundation_.precisionErrorTolerance) {
```

Comment:
can we add a static method in `foundation` with the following signature:

`public static bool floatEquals(float left, float right, float precisionTolerance = precisionErrorTolerance)`

and then use this method to replace all similar expressions in UIWidgets?

Reply:
updated